File: frontend/src/services/notificationService.ts
import { apiClient } from '../api/client';
import { ApiResponse } from '../types';

export interface NotificationResponse {
  id: number;
  title: string;
  message: string;
  isRead: boolean;
  createdAt: string;
}

export const notificationService = {
  getMyNotifications: async (): Promise<ApiResponse<NotificationResponse[]>> => {
    const response = await apiClient.get<ApiResponse<NotificationResponse[]>>('/api/v1/notifications');
    return response.data;
  },

  markAsRead: async (id: number): Promise<ApiResponse<string>> => {
    const response = await apiClient.patch<ApiResponse<string>>(`/api/v1/notifications/${id}/read`);
    return response.data;
  },
};

File: backend/src/main/java/com/csms/notification/service/NotificationService.java
package com.csms.notification.service;

import com.csms.common.response.ApiResponse;
import com.csms.notification.dto.NotificationResponse;

import java.util.List;

public interface NotificationService {
    void sendNotification(Long userId, String title, String message);
    void sendNotificationToRole(String roleName, Long departmentId, String title, String message);
    ApiResponse<List<NotificationResponse>> getMyNotifications(String email);
    ApiResponse<String> markAsRead(Long id, String email);
}
